sdk/maps/maps-search-rest/samples-dev/fuzzySearchBatch.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import MapsSearch, {
+  getLongRunningPoller,
+  createBatchItems,
+  SearchAddressBatchResultOutput,
+  SearchSearchAddressBatch200Response,
+} from "@azure-rest/maps-search";
+import { AzureKeyCredential } from "@azure/core-auth";
+import * as dotenv from "dotenv";
+
+dotenv.config();
+
+/**
+ * @summary This sample demonstrates how to use the fuzzy search batch request with MapsSearchClient & getLongRunningPoller.
+ */
+async function main() {
+  /**
+   * Azure Maps supports two ways to authenticate requests:
+   * - Shared Key authentication (subscription-key)
+   * - Azure Active Directory (Azure AD) authentication
+   *
+   * In this sample you can put MAPS_SUBSCRIPTION_KEY into .env file to use the first approach or populate
+   * the three AZURE_CLIENT_ID, AZURE_CLIENT_SECRET & AZURE_TENANT_ID variables for trying out AAD auth.
+   *
+   * More info is available at https://docs.microsoft.com/en-us/azure/azure-maps/azure-maps-authentication.
+   */
+
+  /** Shared Key authentication (subscription-key) */
+  const subscriptionKey = process.env.MAPS_SUBSCRIPTION_KEY || "";
+  const credential = new AzureKeyCredential(subscriptionKey);
+  const client = MapsSearch(credential);
+
+  /** Azure Active Directory (Azure AD) authentication */
+  // const credential = new DefaultAzureCredential();
+  // const mapsClientId = process.env.MAPS_CLIENT_ID || "";
+  // const client = MapsSearch(credential, mapsClientId);
+
+  /** Create batch items with an array of objects which accept the same properties as the fuzzy-search endpoint. */
+  const batchItems = createBatchItems([
+    {
+      query: "atm",
+      lat: 48.858561,
+      lon: 2.294911,
+      radius: 5000,
+      limit: 5,
+    },
+    {
+      query: "Statue Of Liberty",
+      countrySet: ["us"],
+      limit: 2,
+    },
+    {
+      query: "Starbucks",
+      lat: 47.621028,
+      lon: -122.34817,
+      radius: 5000,
+    },
+    // This is an invalid request. We will show how to handle the error later.
+    {
+      query: "atm",
+      lat: 48.858561,
+      lon: 200.294911,
+    },
+  ]);
+
+  /** Create the long running poller with the initial response & MapsSearchClient */
+  const initialResponse = await client
+    .path("/search/fuzzy/batch/{format}", "json")
+    .post({ body: { batchItems } });
+  const poller = getLongRunningPoller(client, initialResponse);
+  /** If the operation is not completed yet, log the partial result we got now and keep polling. */
+  while (!poller.getOperationState().isCompleted) {
+    await poller.poll();
```

Comment:
Is manual polling a common scenario for your users? If so, I think this is good. If you expect most of your users to just care about the final result I would recommend just doing `await poller.pollUntilDone()` so that most users would follow that rather than the manual polling